### PR TITLE
[SPARK-37902][PYTHON] Resolve typing issues detected by mypy==0.931

### PR DESCRIPTION
**Pull request by zero323**
### What changes were proposed in this pull request?

This PR resolves the following typing issue detected by `mypy==0.931`:

```
python/pyspark/pandas/base.py:879: error: "Sequence[Any]" has no attribute "tolist"  [attr-defined]
python/pyspark/sql/tests/test_pandas_udf_typehints_with_future_annotations.py:268: error: Incompatible return value type (got "floating[Any]", expected "float")  [return-value]
python/pyspark/sql/tests/test_pandas_udf_typehints.py:265: error: Incompatible return value type (got "floating[Any]", expected "float")  [return-value]
```

### Why are the changes needed?

To enable smooth migration to newer mypy versions.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing unit tests and `dev/lint-python`.

### CODE REVIEW

File: python/pyspark/sql/tests/test_pandas_udf_typehints.py [hunk 1]
```diff
@@ -261,7 +261,7 @@ def plus_one(itr: Iterator[pd.Series]) -> Iterator[pd.Series]:
     def test_group_agg_udf_type_hint(self):
         df = self.spark.range(10).selectExpr("id", "id as v")
 
-        def weighted_mean(v: pd.Series, w: pd.Series) -> float:
+        def weighted_mean(v: pd.Series, w: pd.Series) -> np.float64:
```
Comment (zero323) [inline python/pyspark/sql/tests/test_pandas_udf_typehints.py:264]: This change is required because [matching `average` signature](https://github.com/numpy/numpy/blob/05d908a31c0be9db3177a5e2f5a543cbeca7e4f9/numpy/lib/function_base.pyi#L117-L122) is shape agnostic, and while `np.float64` inherits from `float`, `np.floating` doesn't.

We could also use `np.floating` here, but this seems to capture intent of the code better.

**Comment by zero323** [comment]
cc @ueshin FYI, since you worked on one of these lately.

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/tests/test_pandas_udf_typehints_with_future_annotations.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
 import sys
 import unittest
 from inspect import signature
-from typing import Union, Iterator, Tuple, cast, get_type_hints
+from typing import Any, Union, Iterator, Tuple, cast, get_type_hints
```
Comment (ueshin) [inline python/pyspark/sql/tests/test_pandas_udf_typehints_with_future_annotations.py:22]: unnecessary change?

Comment (itholic) [reply python/pyspark/sql/tests/test_pandas_udf_typehints_with_future_annotations.py:22]: Oh, btw don't we check the unused import by linter ??

**Comment by itholic** [comment]
LGTM except https://github.com/apache/spark/pull/35199#discussion_r784439721

**Review by itholic — APPROVED** [review]

File: python/pyspark/sql/tests/test_pandas_udf_typehints_with_future_annotations.py (hunk 2 above)
Comment (zero323) [reply python/pyspark/sql/tests/test_pandas_udf_typehints_with_future_annotations.py]: We used to, but it seems like excluding F403 in 90003398745bfee78416074ed786e986fcb2c8c\d messed up that. I am looking into this.

**Comment by zero323** [comment]
Merged into master.

Thanks all.
